tests/test_parser.py

```diff
@@ -203,6 +256,55 @@ def test_get_instructions_from_df(self, dfparser, instruction, instr_value,
 
         assert instructions == expected
 
+    @pytest.mark.parametrize('from_value, expect', [
+        (
+            "    ",
+            (None, None),
+        ), (
+            "   foo",
+            ('foo', None),
+        ), (
+            "foo:bar as baz   ",
+            ('foo:bar', 'baz'),
+        ), (
+            "foo as baz",
+            ('foo', 'baz'),
+        ), (
+            "foo and some other junk",  # we won't judge
+            ('foo', None),
+        ), (
+            "registry.example.com:5000/foo/bar:baz",
+            ('registry.example.com:5000/foo/bar:baz', None),
+        )
+    ])
+    def test_image_from(self, from_value, expect):
+        result = image_from(from_value)
+        assert result == expect
+
+    def test_parent_images(self, dfparser):
+        FROM = ('my-builder:latest', 'rhel7:7.5')
+        template = dedent("""\
+            FROM {0} AS builder
+            CMD do some stuff
+
+            FROM {1}
+            COPY --from=builder some stuff
+            """)
+        dfparser.content = template.format(*FROM)
+
+        parents = dfparser.parent_images
+        for img in FROM:
+            assert img in parents
```

Comment:
How come we're not specifying the order they should appear?

Reply:
Thanks, I didn't update this with the implementation